### PR TITLE
js-application-analytics - Updated to SPFX 1.20

### PR DESCRIPTION
**Pull request by NicolasKheirallah**
> By submitting this pull request, you agree to the [contribution guidelines](https://github.com/pnp/sp-dev-fx-extensions/blob/main/CONTRIBUTING.md)

> If you aren't familiar with how to contribute to open-source repositories using GitHub, or if you find the instructions on this page confusing, [sign up](https://forms.office.com/Pages/ResponsePage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUREZVRDVYUUJLT1VNRDM4SjhGMlpUNzBORy4u) for one of our [Sharing is Caring](https://pnp.github.io/sharing-is-caring/#pnp-sic-events) events. It's completely free, and we'll guide you through the process.

|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | no                              |
| New feature?    | no                               |
| New sample?     | no                             |
| Related issues? | Updated to SPFX 1.20 and updated rest of modules etc. |

## What's in this Pull Request?

> Please describe the changes in this PR. Sample description or details around bugs which are being fixed.
Updated to SPFX 1.20 and updated rest of modules  etc

### CODE REVIEW

File: samples/js-application-analytics/README.md
Comment (Adam-it) [inline samples/js-application-analytics/README.md:1]: in this file in the `Compatibility` section you should also update to tags to SPFx 1.20 and Node to v18

File: samples/js-application-analytics/package.json [hunk 1]
```diff
@@ -12,22 +12,23 @@
...
-    "@microsoft/sp-webpart-workbench": "1.11.0",
-    "@microsoft/rush-stack-compiler-3.3": "0.3.5",
-    "gulp": "~3.9.1",
-    "@types/chai": "3.4.34",
-    "@types/mocha": "2.2.38",
-    "ajv": "~6.12.3",
-    "@types/webpack-env": "1.13.1",
-    "@types/es6-promise": "0.0.33"
+    "@microsoft/rush-stack-compiler-3.3": "0.11.2",
+    "@microsoft/rush-stack-compiler-4.5": "^0.5.1",
```
Comment (Adam-it) [inline samples/js-application-analytics/package.json:22]: seems we have two versions of rush-stack now. 
only the second one should stay

File: samples/js-application-analytics/src/extensions/analytics/AnalyticsApplicationCustomizer.tsx
Comment (Adam-it) [inline samples/js-application-analytics/src/extensions/analytics/AnalyticsApplicationCustomizer.tsx:1]: don't we have the same in `.ts` file? why we duplicate the logic?

**Review by Adam-it — CHANGES_REQUESTED** [review]
@NicolasKheirallah Awesome work so far 👏👏👏
I didn't manage to `gulp bundle` the solution. Lets double check if it build before we merge
image

Also besides the files you already change we should also update the `assets/sample.json` and there we should update the `updateDateTime` and `SPFX-VERSION` metadata to 1.20.0.

Also did you generate the upgrade guidance for this project and go over all the required changes? I did that and a few recommended and required things that still should be done before we proceed 
image

File: samples/js-application-analytics/src/extensions/analytics/AnalyticsApplicationCustomizer.tsx
Comment (NicolasKheirallah) [reply samples/js-application-analytics/src/extensions/analytics/AnalyticsApplicationCustomizer.tsx:1]: Thanks for the feedback! Learning alot! :) 
Yeah, It was suppose only to have a .TS file
All suggested issues should be fixed now! :)

File: samples/js-application-analytics/README.md
Comment (Adam-it) [reply samples/js-application-analytics/README.md:1]: @NicolasKheirallah I think you still missed to update the tags in the readme file. Could you have a recheck?

Comment (NicolasKheirallah) [reply samples/js-application-analytics/README.md:1]: You are 100% correct! Tags have now been updated! :)
image

**Review by Adam-it — APPROVED** [review]
@NicolasKheirallah awesome work 👏👏👏
LGTM 👍

**Pull request merged**